kof-operator/internal/controller/clusterdeployment_utils.go

```diff
@@ -0,0 +1,16 @@
+package controller
+
+import (
+	kcmv1alpha1 "github.com/K0rdent/kcm/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func GetOwnerReference(ownerName string, ownerUID types.UID) metav1.OwnerReference {
+	return metav1.OwnerReference{
+		APIVersion: kcmv1alpha1.GroupVersion.String(),
+		Kind:       kcmv1alpha1.ClusterDeploymentKind,
+		Name:       ownerName,
+		UID:        ownerUID,
+	}
+}
```

Comment:
Part 2 of DRY-ing it:
```suggestion
func GetOwnerReference(owner client.Object) metav1.OwnerReference {
	gvk := owner.GetObjectKind().GroupVersionKind()
	return metav1.OwnerReference{
		APIVersion: gvk.GroupVersion().String(),
		Kind:       gvk.Kind,,
		Name:       ownerName,
		UID:        ownerUID,
	}
}
```

Reply:
Fixed typos above

Reply:
Tests say `APIVersion` and `Kind` are empty strings, checking why...

Reply:
@AndrejsPon00 looks like it is caused by how we create our test objects. Real objects will have gvk, but in our fake test objects a lot of fields are missing, so we have to add gvk there manually. Please add this to the `createClusterDeployment` function:
```
clusterDeployment.SetGroupVersionKind(schema.GroupVersionKind{
	Group:   "k0rdent.mirantis.com",
	Version: "v1alpha1",
	Kind:    "ClusterDeployment",
})
```

Reply:
Looks like even typed object created manually like we do may still have no fields required for GVK.
Found one more way, more concise:
```
			clusterDeployment := &kcmv1alpha1.ClusterDeployment{
						TypeMeta: metav1.TypeMeta{
									APIVersion: "k0rdent.mirantis.com/v1alpha1",
									Kind:       "ClusterDeployment",
						},
						ObjectMeta: metav1.ObjectMeta{
```
Please try it instead of `SetGroupVersionKind`.

Reply:
In Kubernetes, there is a known issue where `TypeMeta` fields are intentionally cleared during object decoding. This is most likely why tests don't work.
https://github.com/kubernetes/kubernetes/issues/80609
https://github.com/kubernetes-sigs/controller-runtime/issues/1870